framework/src/modules/chain/bft/finality_manager.js

```diff
@@ -117,6 +117,25 @@ class FinalityManager extends EventEmitter {
 		return this;
 	}
 
+	/**
+	 * Remove block headers above given height (exclusive)
+	 * If no param provided will remove last block header
+	 *
+	 *
+	 * @param {int} aboveHeight -  Height from above remove headers
+	 */
+	removeBlockHeaders({ aboveHeight }) {
+		debug('removeBlockHeaders invoked');
```

Comment:
Is this something new we started using in the application?

Reply:
We have been using this in http API, but general consensus is this can be used in lowest layer